File: 1802-number-of-students-unable-to-eat-lunch/1802-number-of-students-unable-to-eat-lunch.py
class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        res = len(students)
        dict1 = {}

        for i in students:
            if i not in dict1:
                dict1[i] = 1
            else:
                dict1[i] += 1

        for s in sandwiches:
            if dict1.get(s, 0) > 0:
                res -= 1
                dict1[s] -= 1
            else:
                break

        return res

File: 1802-number-of-students-unable-to-eat-lunch/1802-number-of-students-unable-to-eat-lunch.java
class Solution {
    public int countStudents(int[] students, int[] sandwiches) {
        int res = students.length;
        HashMap<Integer, Integer> dict1 = new HashMap<>();

        for (int student : students) {
            dict1.put(student, dict1.getOrDefault(student, 0) + 1);
        }

        for (int s : sandwiches) {
            if (dict1.getOrDefault(s, 0) > 0) {
                res--;
                dict1.put(s, dict1.get(s) - 1);
            } else {
                break;
            }
        }

        return res;
    }
}